### ISSUE TITLE
Attention op (opset 24): add optional pre-softcap additive bias input

### ISSUE THREAD

**Issue by justinchuby** [onnx/onnx#7829]
## Summary

The ONNX `Attention` operator (opset 24) has a fixed internal pipeline:

```
QK matmul → scale → softcap → attn_mask add → softmax → weighted sum
```

There is no way to inject an **additive bias** (e.g. a relative position bias) **after** the QK matmul but **before** the softcap. This makes it impossible to use the native `softcap` attribute for models that require the order:

```
QK matmul → scale → relative_position_bias add → softcap → attn_mask add → softmax
```

## Concrete use case: Gemma 4 audio encoder

The Gemma 4 audio Conformer attention ([`Gemma4ConformerAttention`](https://huggingface.co/google/gemma-4-E2B-it)) computes:

1. Standard scaled QK scores
2. **Adds a learnable relative position bias** (`rel_bias[b, h, i, j]`) to the scores
3. **Applies softcapping**: `tanh((qk + rel_bias) / cap) * cap`
4. Adds a causal sliding-window mask
5. Softmax

Because step 3 must see `qk + rel_bias` together, and the ONNX `Attention` op applies `softcap` before any mask/bias addition, the native `softcap` attribute cannot be used. The workaround is to implement the full attention pipeline manually using lower-level ops (`MatMul`, `Tanh`, `Div`, `Mul`, `Softmax`, etc.), which:

- Prevents use of hardware-accelerated fused attention kernels
- Increases graph complexity and reduces portability

## Current spec

The `softcap` attribute is documented at:
https://onnx.ai/onnx/operators/onnx__Attention.html

Its placement in the pipeline is fixed; there is no input for a pre-softcap additive term.

## Proposed change

Add an optional input to the `Attention` op — tentatively named **`attention_bias`** — with shape `[batch, num_heads, seq, seq]` (or broadcastable equivalent). This bias would be applied **after** the QK matmul and scale, but **before** softcap and before the existing `attn_mask`:

```
QK matmul → scale → attention_bias add → softcap → attn_mask add → softmax
```

This mirrors the convention used by PyTorch [`scaled_dot_product_attention`](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) (its `attn_bias` parameter) and FlashAttention-2's `alibi_slopes`/bias inputs.

### Alternative

If a single ordering cannot satisfy all use cases, a boolean attribute (e.g. `bias_before_softcap`) could control whether `attention_bias` is applied before or after softcap.

## Impact

Models affected include any architecture that combines:
- Relative position biases (ALiBi, T5-style, Conformer, Gemma 4 audio)
- Attention logit soft-capping (Gemma family)

Without this, implementers must fall back to manual attention graphs, losing the performance benefits of fused `Attention` kernels on all backends.

## References

- ONNX `Attention` op spec: https://onnx.ai/onnx/operators/onnx__Attention.html
- Gemma 4 audio encoder (HuggingFace): https://huggingface.co/google/gemma-4-E2B-it
- PyTorch `scaled_dot_product_attention` `attn_bias`: https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html

**Comment by 0xjah** [comment]
hi @justinchuby can i get assigned for this issue
